File: ports/1.21.1-full/src/main/java/mods/eln/misc/FakeSideInventory.java
package mods.eln.misc;

import net.minecraft.world.entity.player.Player;
import net.minecraft.world.WorldlyContainer;
import net.minecraft.world.item.ItemStack;
import net.minecraft.core.Direction;
import net.minecraft.network.chat.Component;

public class FakeSideInventory implements WorldlyContainer {

    static private final FakeSideInventory instance = new FakeSideInventory();

    public static FakeSideInventory getInstance() {
        return instance;
    }

    @Override
    public int getContainerSize() {
        return 0;
    }

    @Override
    public ItemStack getItem(int var1) {
        return ItemStack.EMPTY;
    }

    @Override
    public ItemStack removeItem(int var1, int var2) {
        return ItemStack.EMPTY;
    }

    @Override
    public ItemStack removeItemNoUpdate(int var1) {
        return ItemStack.EMPTY;
    }

    @Override
    public void setItem(int var1, ItemStack var2) {
    }

    @Override
    public boolean isEmpty() {
        return true;
    }

    
    public String getName() {
        return "FakeSideInventory";
    }

    
    public boolean hasCustomName() {
        return false;
    }

    
    public Component getDisplayName() {
        return Component.literal(getName());
    }

    @Override
    public int getMaxStackSize() {
        return 0;
    }

    @Override
    public void setChanged() {
    }

    @Override
    public boolean stillValid(Player player) {
        return false;
    }

    @Override
    public void startOpen(Player var1) {

    }

    @Override
    public void stopOpen(Player var1) {

    }

    @Override
    public boolean canPlaceItem(int var1, ItemStack var2) {
        return false;
    }

    
    public int getField(int id) {
        return 0;
    }

    
    public void setField(int id, int value) {

    }

    
    public int getFieldCount() {
        return 0;
    }

    @Override
    public void clearContent() {

    }

    @Override
    public int[] getSlotsForFace(Direction var1) {
        return new int[]{};
    }

    @Override
    public boolean canPlaceItemThroughFace(int var1, ItemStack var2, Direction var3) {
        return false;
    }

    @Override
    public boolean canTakeItemThroughFace(int var1, ItemStack var2, Direction var3) {
        return false;
    }
}

File: src/main/kotlin/mods/eln/misc/FakeSideInventory.kt
package mods.eln.misc

import net.minecraft.world.entity.player.Player
import net.minecraft.world.WorldlyContainer
import net.minecraft.world.item.ItemStack
import net.minecraft.core.Direction
import net.minecraft.network.chat.Component

class FakeSideInventory : WorldlyContainer {
    override fun getContainerSize(): Int {
        return 0
    }

    override fun isEmpty(): Boolean = true

    override fun getItem(var1: Int): ItemStack {
        return ItemStack.EMPTY
    }

    override fun removeItem(var1: Int, var2: Int): ItemStack {
        return ItemStack.EMPTY
    }

    override fun removeItemNoUpdate(var1: Int): ItemStack {
        return ItemStack.EMPTY
    }

    override fun setItem(var1: Int, var2: ItemStack) {}
    override fun getName(): String {
        return "FakeSideInventory"
    }

    override fun hasCustomName(): Boolean {
        return false
    }

    override fun getDisplayName(): Component = Component.literal(name)

    override fun getMaxStackSize(): Int {
        return 0
    }

    override fun setChanged() {}
    override fun stillValid(var1: Player): Boolean {
        return false
    }

    override fun startOpen(player: Player) {}
    override fun stopOpen(player: Player) {}
    override fun canPlaceItem(var1: Int, var2: ItemStack): Boolean {
        return false
    }

    override fun getField(id: Int): Int = 0
    override fun setField(id: Int, value: Int) {}
    override fun getFieldCount(): Int = 0
    override fun clear() {}

    override fun getSlotsForFace(side: Direction): IntArray {
        return intArrayOf()
    }

    override fun canPlaceItemThroughFace(index: Int, stack: ItemStack, direction: Direction?): Boolean {
        return false
    }

    override fun canTakeItemThroughFace(index: Int, stack: ItemStack, direction: Direction): Boolean {
        return false
    }

    companion object {
        @JvmStatic
        val instance = FakeSideInventory()
    }
}
